app/services/katello/registration_manager.rb

```diff
@@ -297,7 +298,7 @@ def remove_host_artifacts(host, clear_content_facet = true)
         host.get_status(::Katello::TraceStatus).destroy
         host.installed_packages.delete_all
 
-        host.rhsm_fact_values.destroy_all
+        host.rhsm_fact_values.delete_all
```

Comment:
I don't mind reverting this, but thought it was simple enough to include. Changed to `delete_all` since we aren't concerned with the values that are destroyed